### PR TITLE
Fix #16425 - Add DELETE FROM table

### LINKED ISSUE #16425: Delete from $table
Table operations tab has a `truncate $table` operation but not a `delete from $table` operation. Could it be added?
Truncate doesn't work on tables with foreign keys.

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Please describe your pull request.

Fixes #16425 

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Controllers/JavaScriptMessagesController.php [hunk 1]
```diff
@@ -42,6 +42,7 @@ private function setMessages(): void
             ),
             'strDropTableStrongWarning' => __('You are about to DESTROY a complete table!'),
             'strTruncateTableStrongWarning' => __('You are about to TRUNCATE a complete table!'),
+            'strDeleteTableStrongWarning' => __('You are about to DELETE a complete table!'),
```
Comment (williamdes) [inline libraries/classes/Controllers/JavaScriptMessagesController.php:45]:
```suggestion
            'strDeleteTableStrongWarning' => __('You are about to DELETE all the rows of the table'),
```

**Comment by williamdes** [comment]
@sudo-robot deploy

File: js/src/table/operations.js [hunk 2]
```diff
@@ -10,8 +10,48 @@ AJAX.registerTeardown('table/operations.js', function () {
     $(document).off('click', '#drop_tbl_anchor.ajax');
     $(document).off('click', '#drop_view_anchor.ajax');
     $(document).off('click', '#truncate_tbl_anchor.ajax');
+    $(document).off('click', '#delete_tbl_anchor.ajax');
 });
 
+var confirmAndPost = function (obj, action) {
```
Comment (williamdes) [inline js/src/table/operations.js:16]:
```suggestion
/**
  * @param {Object} obj
  * @param {'TRUNCATE'|'DELETE'} action
  */
var confirmAndPost = function (obj, action) {
```

**Review by williamdes — APPROVED** [review]
Nice work !

**Comment by williamdes** [comment]
I will merge this one soon
@OlafvdSpek if you want to review ? (https://github.com/phpmyadmin/phpmyadmin/pull/16750#issuecomment-803594484)

**Comment by OlafvdSpek** [comment]
The confirmation says `"DELETE `public`.`test1`"`, which isn't correct. Ideally the query for the confirmation and the actual query would be build by the same code..

image

Not sure if this is a problem..

**Comment by shucon** [comment]
> The confirmation says
> 
> ```
> "DELETE `public`.`test1`"
> ```
> 
> which isn't correct. Ideally the query for the confirmation and the actual query would be build by the same code.. 
> Not sure if this is a problem..

Thanks for the suggestion @OlafvdSpek . I've fixed it for `DROP` and `TRUNCATE` query as well.

**Comment by shucon** [comment]
@sudo-robot deploy

**Comment by williamdes** [comment]
I pushed 21566d4bb59d2a02716447213c2282986fd64b69 to QA to extract the fix added here

**Review by williamdes — APPROVED** [review]
Works great, I did some code improvements on top of your changes and rebased them

**Pull request merged**